Assets/Scripts/Diablerie/Engine/IO/Wav.cs

```diff
@@ -130,6 +130,19 @@ private static float ReadSample(BinaryReader reader)
         return BytesToFloat(byte1, byte2);
     }
 
+    private static void ReadSample(BinaryReader reader, float[] sampleBuffer, int startIndex, int sampleCount)
+    {
+        byte[] buffer = new byte[sampleCount * 2];
```

Comment:
Could we reuse the buffer array somehow to avoid frequent allocations and excessive garbage collector work?

Reply:
How about create a reuseable fixed length byte[] buffer when `new Wav()`, and transfer this buffer to ReadSample method. Works like a Stream's internal buffer.

Reply:
Sounds good